changelog/unreleased/ocm-model-refactor.md

```diff
@@ -0,0 +1,24 @@
+Enhancement: refactoring of the GORM model for shares
+
+With this PR we introduce new constraints and rename
+some fields for better consistency:
+
+* Types used by OCM structures only are prefixed with
+  `Ocm`, and `AccessMethod` and `Protocol` were
+  consolidated into `OcmProtocol`
+* ItemType is used in OCM shares as well
+* `Type` has been renamed to `RecipientType` with
+  type `OcmShareType`, and converted both ways
+* Unique index constraints have been created as follows:
+  * For `Shares` on `(inode, instance, permissions, recipient, deletedAt)`
```

Comment:
Why are permissions part of the index? One recipient should have at most 1 share, so it's not dependent on the permissions.

Reply:
Right! Will fix